adapters/repos/db/index.go

```diff
@@ -557,12 +558,21 @@ func (i *Index) updateInvertedIndexConfig(ctx context.Context,
 	return nil
 }
 
+func (i *Index) overrideAsyncReplication() bool {
+	// Disable async replication regardless of class setting if env var is set
+	return entcfg.Enabled(os.Getenv("DISABLE_ASYNC_REPLICATION"))
+}
+
 func (i *Index) updateAsyncReplication(ctx context.Context, enabled bool) error {
 	i.asyncReplicationLock.Lock()
 	defer i.asyncReplicationLock.Unlock()
 
 	i.Config.AsyncReplicationEnabled = enabled
 
+	if i.overrideAsyncReplication() {
+		i.Config.AsyncReplicationEnabled = false
```

Comment:
I think I would read the env variable in `func (i *Index) asyncReplicationEnabled() bool`  instead